packages/amplify_core/lib/src/types/auth/attribute/cognito_user_attribute_key.dart

```diff
@@ -16,17 +16,20 @@ import 'package:meta/meta.dart';
 class CognitoUserAttributeKey extends AuthUserAttributeKey
```

Comment:
why do we have Cognito types in core? I recommend moving the category level behaviors to `AuthUserAttributeKey` and those that are specific to Cognito out of core and into the Auth package.

Reply:
The Cognito configuration (and all plugin configurations) are in `core` and this is needed as part of it.

> category level behaviors to AuthUserAttributeKey

This has already been done unless I missed something?

Reply:
this is right, I pulled the PR and looked at the code in details. now I understand the structure and how config is done in core package.